rust/worker/src/execution/operators/apply_log_to_segment_writer.rs

```diff
@@ -91,22 +81,25 @@ impl WriteSegmentsInput {
 }
 
 #[derive(Debug)]
-pub struct WriteSegmentsOutput {
-    pub(crate) record_segment_writer: RecordSegmentWriter,
-    pub(crate) hnsw_segment_writer: Box<DistributedHNSWSegmentWriter>,
-    pub(crate) metadata_segment_writer: MetadataSegmentWriter<'static>,
-}
+pub struct ApplyLogToSegmentWriterOutput {}
 
 #[async_trait]
-impl Operator<WriteSegmentsInput, WriteSegmentsOutput> for WriteSegmentsOperator {
-    type Error = WriteSegmentsOperatorError;
+impl<Writer: SegmentWriter + Send + Sync + Clone>
+    Operator<ApplyLogToSegmentWriterInput<Writer>, ApplyLogToSegmentWriterOutput>
+    for ApplyLogToSegmentWriterOperator
+{
+    type Error = ApplyLogToSegmentWriterOperatorError;
 
     fn get_name(&self) -> &'static str {
         "WriteSegmentsOperator"
```

Comment:
nit- this name has changed

Reply:
yeah, realized this last night as well haha
thanks